services/pub/pub-popularity.service.js

```diff
@@ -0,0 +1,56 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+
+const documentation = `<p>A measure of how many developers use a package, providing insight into what other developers are using.</p>`
+
+const keywords = ['dart', 'flutter']
+
+const schema = Joi.object({
+  popularityScore: Joi.number().min(0).max(1).required(),
+}).required()
+
+const title = 'Pub Popularity'
+
+export default class PubPopularity extends BaseJsonService {
+  static category = 'rating'
+
+  static route = { base: 'pub/popularity', pattern: ':packageName' }
+
+  static examples = [
+    {
+      title,
+      keywords,
+      documentation,
+      namedParams: { packageName: 'analysis_options' },
+      staticPreview: {
+        label: 'popularity',
+        message: '100%',
+        color: 'brightgreen',
+      },
+    },
+  ]
+
+  static defaultBadgeData = { label: 'popularity' }
+
+  static render({ popularityScore, packageName }) {
+    return {
+      label: 'popularity',
+      message: `${Math.round(popularityScore * 100)}%`,
+      color: 'brightgreen',
```

Comment:
As with #7918 there is the opportunity to convey some useful information using color here rather than just make everything `brightgreen` even if your score is low. Same question here really. Is there any existing community norm for what represents a "good" score we can represent here?

Reply:
For now, I have added the blue color for popularity scores as there are no such norms for scores.

Reply:
OK. Most of our rating badges are for star ratings (out of 5). Translating the splits we use there to percentages would give us 

```js
const roundedScore = Math.round(popularityScore * 100)

return {
  label: 'popularity',
  message: `${roundedScore}%`,
  color: floorCount(roundedScore, 40, 60, 80),
}
```

I reckon lets go with that. It seems like popularity is a pretty straightforward scale and this will be consistent with other rating badges.

Reply:
Done.